optimum/bettertransformer/models/encoder_models.py

```diff
@@ -1192,3 +1191,115 @@ def _get_activation_function(self, config: "PretrainedConfig"):
+        """
+        super().__init__(config)
+        # In_proj layer
+        self.in_proj_weight = nn.Parameter(
+            torch.cat(
+                [
+                    prnt_layer.self_attn.query_proj.weight,
+                    prnt_layer.self_attn.key_proj.weight,
+                    prnt_layer.self_attn.value_proj.weight,
+                ]
+            )
+        )
+        self.in_proj_bias = nn.Parameter(
+            torch.cat(
+                [
+                    prnt_layer.self_attn.query_proj.bias,
+                    prnt_layer.self_attn.key_proj.bias,
+                    prnt_layer.self_attn.value_proj.bias,
+                ]
+            )
+        )
+
+        # Out proj layer
+        self.out_proj_weight = prnt_layer.self_attn.out_proj.weight
+        self.out_proj_bias = prnt_layer.self_attn.out_proj.bias
+
+        # Linear layer 1
+        self.linear1_weight = prnt_layer.feed_forward.intermediate.weight
+        self.linear1_bias = prnt_layer.feed_forward.intermediate.bias
+
+        # Linear layer 2
+        self.linear2_weight = prnt_layer.feed_forward.output.weight
+        self.linear2_bias = prnt_layer.feed_forward.output.bias
+
+        # Layer norm 1
+        self.norm1_eps = prnt_layer.self_attn_layer_norm.eps
+        self.norm1_weight = prnt_layer.self_attn_layer_norm.weight
+        self.norm1_bias = prnt_layer.self_attn_layer_norm.bias
+
+        # Layer norm 2
+        self.norm2_eps = prnt_layer.feed_forward_layer_norm.eps
+        self.norm2_weight = prnt_layer.feed_forward_layer_norm.weight
+        self.norm2_bias = prnt_layer.feed_forward_layer_norm.bias
+
+        # Model hyper parameters
+        self.num_heads = prnt_layer.self_attn.num_attn_heads
+        self.embed_dim = prnt_layer.self_attn.head_dim
+
+        # Last step: set the last layer to `False` -> this will be set to `True` when converting the model
+        self.is_last_layer = False
+
+        self.validate_bettertransformer()
+
+    def forward(self, hidden_states, attention_mask, position_bias=None, *_, **__):
+        r"""
+        This is just a wrapper around the forward function proposed in:
+        https://github.com/huggingface/transformers/pull/19553
+        """
+        super().forward_checker()
+
+        if hidden_states.is_nested:
+            attention_mask = None
+
+        if attention_mask is not None:
+            # attention mask comes in with values 0 and -inf. we convert to torch.nn.TransformerEncoder style bool mask
+            # 0->false->keep this token -inf->true->mask this token
+            attention_mask = attention_mask.bool()
+            attention_mask = torch.reshape(attention_mask, (attention_mask.shape[0], attention_mask.shape[-1]))
+            seqlen = attention_mask.shape[1]
+            lengths = torch.sum(~attention_mask, 1)
+
+            if hidden_states.shape[0] != attention_mask.shape[0]:
+                hidden_states = hidden_states.transpose(1, 0)
```

Comment:
Maybe the error you get is due to those, not sure.

Reply:
I just tried that.. no luck, those tests are still failing. Just wondering, what is the purpose of the above code snippet?

Reply:
By using print statements, I can see that hidden_states is of shape: torch.Size([2, 4, 16]) and attention_mask is of shape: torch.Size([8, 4]). So there appears to be a discrepancy in the batch_size value?

Reply:
Hi @adit299 I'll look asap, probably on Friday. In the meanwhile you can compare with the shapes for e.g. Bert to see how it differs.